gef-ui/server/controllers/unissued-facilities/facility-end-date/index.ts

```diff
@@ -0,0 +1,217 @@
+import { Response } from 'express';
+import { getDate, getMonth, getYear, isSameDay, parseISO, startOfDay } from 'date-fns';
+import { CustomExpressRequest, isFacilityEndDateEnabledOnGefVersion, parseDealVersion } from '@ukef/dtfs2-common';
+import { isTrueSet, validationErrorHandler } from '../../../utils/helpers';
+import * as api from '../../../services/api';
+import { validateAndParseFacilityEndDate } from './validation';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { FacilityEndDateViewModel } from '../../../types/view-models/facility-end-date-view-model';
+import { ValidationError } from '../../../types/validation-error';
+
+type FacilityEndDateParams = { dealId: string; facilityId: string };
+type FacilityEndDatePostBody = { 'facility-end-date-day': string; 'facility-end-date-month': string; 'facility-end-date-year': string };
+
+type GetFacilityEndDateRequest = CustomExpressRequest<{ params: FacilityEndDateParams; query: { change: string } }>;
+type PostFacilityEndDateRequest = CustomExpressRequest<{
+  reqBody: FacilityEndDatePostBody;
+  params: FacilityEndDateParams;
+  query: { saveAndReturn: string };
+}>;
+
+const getFacilityEndDateViewModel = async (req: GetFacilityEndDateRequest, previousPage: string): Promise<FacilityEndDateViewModel | null> => {
```

Comment:
I'm not sure I'm keen on this returning null pattern, but it's relatively common so let's keep this for now.